src/test/java/org/gridsuite/study/server/StudyTest.java

```diff
@@ -693,6 +694,21 @@ public void testSearch() throws Exception {
                 .andExpectAll(status().isBadRequest(),
                         content().string("Enum unknown entry 'bogus' should be among NAME, ID"))
                 .andReturn();
+
+        // Create a line modification with a specific variantId, then perform a search for this modified entry.
+        // This process should return a unique result for each equipment, thereby ensuring no duplicates are present in the search results.
+
+        NetworkModificationNode node1 = createNetworkModificationNode(studyUuid, rootNodeId, VARIANT_ID, "node1", "userId");
```

Comment:
I don't undersand your test
I think it's not working

Reply:
It's working. I added this test because the new process I put in was not covered in the current tests

Reply:
To make the test more clear, in my opinion, we can rectify test as following 

`        // Create a line modification with a specific variantId, then perform a search for this modified entry.
        // This process should return a unique result for each equipment, thereby ensuring no duplicates are present in the search results.
        NetworkModificationNode node1 = createNetworkModificationNode(studyUuid, rootNodeId, VARIANT_ID, "node1", "userId");

        List<EquipmentInfos> mockLineInfosWithModification = new ArrayList<>(linesInfos);
        // get whatever initial line to mock a line modification, here we take the first one as an example
        EquipmentInfos lineInitial = mockLineInfosWithModification.get(0);
        EquipmentInfos lineModification = EquipmentInfos.builder()
                .networkUuid(NETWORK_UUID)
                .id(lineInitial.getId())
                .name(lineInitial.getName())
                .type(lineInitial.getType())
                .variantId(VARIANT_ID)
                .voltageLevels(lineInitial.getVoltageLevels())
                .build();

        mockLineInfosWithModification.add(lineModification);

        // prepare expected result
        List<EquipmentInfos> expectedLineInfosWithModification = new ArrayList<>(mockLineInfosWithModification);
        expectedLineInfosWithModification.remove(lineInitial);

        // mock return of the equipmentInfosService
        when(equipmentInfosService.searchEquipments(any(BoolQuery.class))).then((Answer<List<EquipmentInfos>>) invocation -> mockLineInfosWithModification);

        mvcResult = mockMvc
                .perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/search?userInput={request}&fieldSelector=NAME",
                        studyUuid, node1.getId(), "L").header(USER_ID_HEADER, "userId"))
                .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON)).andReturn();
        equipmentInfos = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() { });

        assertThat(equipmentInfos, new MatcherJson<>(mapper, expectedLineInfosWithModification));`

Reply:
In my opinion, it is better to extract to a new method, for example testSearchWithModification to make the test more isolated since we override the mock, i.e.  `when(equipmentInfosService.searchEquipments(any(BoolQuery.class))).then((Answer<List<EquipmentInfos>>) invocation -> mockLineInfosWithModification); `